### ISSUE TITLE
File write/delete inconsistencies

### ISSUE THREAD

**Issue by petteriTeikari** [awslabs/mountpoint-s3#570]
## Mountpoint for Amazon S3 version
mount-s3 1.0.2

## AWS Region
eu-north-1

## Describe the running environment
* Local: Ubuntu 22.04
* Docker: 24.0.6

## What happened?

Sorry if this question is a bit too _stackoverflowish_ as it could be just that my Docker skills are not enough for the task, but I cannot figure out how to even get to the source of this problem.

I want to mount a S3 bucket so that I can write artifacts to it.

I can do this on my local laptop and it works as intended (I can write to the bucket and delete files from Python script):

```
mount-s3 --region eu-north-1  --allow-delete --allow-other  bucket-artifacts artifacts
```

<details>
  <summary>Permission log ([Python code](https://github.com/petteriTeikari/minivess_mlops/blob/7bbd63673be749b0528e77cbac015d0f8c0900af/src/utils/config_utils.py#L407)): Local</summary>

  ```
  MOUNT: /home/petteri/artifacts
   owned by petteri:petteri (owner:group)
   owned by 1000:1000 (owner:group)
   mount permissions: 0755
   read access = True
   write access = True
   execution access = True
   existence of dir = True
  Trying to write to the mount (write access was OK)
   File write succesful!
   file_permission = 0644, 1000:1000
   File delete succesful!
  ```

</details>

And then when running the Python script from Docker container ([base](https://github.com/petteriTeikari/minivess_mlops/blob/7bbd63673be749b0528e77cbac015d0f8c0900af/deployment/Dockerfile_env)+[app](https://github.com/petteriTeikari/minivess_mlops/blob/7bbd63673be749b0528e77cbac015d0f8c0900af/deployment/Dockerfile)) mounting in the [entrypoint.sh](https://github.com/petteriTeikari/minivess_mlops/blob/7bbd63673be749b0528e77cbac015d0f8c0900af/deployment/entrypoint.sh)), I can write the test_file, but cannot delete it right after:

<details>
  <summary>Permission log: Run 1</summary>

  ```
MOUNT: /mnt/minivess-artifacts
 owned by minivessuser:minivessuser (owner:group)
 owned by 1000:1000 (owner:group)
 mount permissions: 0755
 read access = True
 write access = True
 execution access = True
 existence of dir = True
Trying to write to the mount (write access was OK)
 File write succesful!
 file_permission = 0644, 1000:1000
 Problem deleting file /mnt/minivess-artifacts/test_write.txt, e = [Errno 1] Operation not permitted: '/mnt/minivess-artifacts/test_write.txt'
  ```
![1](https://github.com/awslabs/mountpoint-s3/assets/1060514/1655d717-6907-4348-8c7d-88305342a8c6)

</details>

And then weirdly when running the same script, I can delete the file created by the previous run, and then write the same file again, but again not being able to delete it right after? 

<details>
  <summary>Permission log: Run 2</summary>

  ```
MOUNT: /mnt/minivess-artifacts
 owned by minivessuser:minivessuser (owner:group)
 owned by 1000:1000 (owner:group)
 mount permissions: 0755
 read access = True 
 write access = True
 execution access = True
 existence of dir = True
Trying to write to the mount (write access was OK)
 File /mnt/minivess-artifacts/test_write.txt already exists, deleting it first
 File write succesful!
 file_permission = 0644, 1000:1000
 Problem deleting file /mnt/minivess-artifacts/test_write.txt, e = [Errno 1] Operation not permitted: '/mnt/minivess-artifacts/test_write.txt'
```

![2](https://github.com/awslabs/mountpoint-s3/assets/1060514/624a3342-9fda-4d8c-b0bf-7f51f866157c)

</details>

* Am I trying to delete it too fast after initial write?
* Is there some way to check from Python that the mount is "working ok" (as in how it functions from laptop directly)?
* Or should I just settle with [`aws sync`](https://docs.aws.amazon.com/cli/latest/reference/s3/sync.html) after the creation of all the artifacts and not try to write as they are generated?

**Comment by passaro** [comment]
About your first question:
> Am I trying to delete it too fast after initial write?

Likely. In Mountpoint 1.0.2 the upload may not yet be complete when a file is closed, and that could cause the delete immediately after to fail. A recently merged change (#526) should address this problem and will be included in the next release.
I would suggest retrying with the new Mountpoint release when it is out (or with a build from `main` if feasible).

**Comment by jamesbornholt** [comment]
Just a note that [Mountpoint v1.1.0](https://github.com/awslabs/mountpoint-s3/releases/tag/mountpoint-s3-1.1.0) is out and contains the fix @passaro mentioned, if you'd like to test again to see if the problem is resolved.

**Comment by dannycjones** [comment]
I'll close this now as we believe this should be resolved with [Mountpoint v1.1.0](https://github.com/awslabs/mountpoint-s3/releases/tag/mountpoint-s3-1.1.0). Please do re-open or open a new issue if needed.
